### PR TITLE
new: usr: Censys Expansion module

### LINKED ISSUE #384: censys.io expansion module
This feature request is for a censys.io expansion module . 

Perhaps the shodan module can be a source of inspiration.

### PR DESCRIPTION
**Pull request by Golbark**
Fixes #384 

Provides a new module for integration with censys.io

### CODE REVIEW

File: misp_modules/modules/expansion/censys_enrich.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    if not request.get('attribute'):
+        return {'error': 'Unsupported input.'}
+    attribute = request['attribute']
+    if not any(input_type == attribute['type'] for input_type in mispattributes['input']):
+        return {'error': 'Unsupported attributes type'}
+
+    attribute = MISPAttribute()
+    attribute.from_dict(**request['attribute'])
+
+    if attribute.type == 'ip-dst' or attribute.type == 'ip-src':
```
Comment (cvandeplas) [inline misp_modules/modules/expansion/censys_enrich.py:47]: Does this work with combined types? like `domain|ip` ?

File: misp_modules/modules/expansion/censys_enrich.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        conn = censys.ipv4.CensysIPv4(api_id=api_id, api_secret=api_secret)
+    elif attribute.type == 'domain':
```
Comment (cvandeplas) [inline misp_modules/modules/expansion/censys_enrich.py:49]: `hostname` , `domain|ip` ?

File: misp_modules/modules/expansion/censys_enrich.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+    This function check if object (json format) is present in list_objects
+    using attribute_name for the matching
+    """
+    for o in list_objects:
+        if o['name'] == object['name']:
+            for attr in object['Attribute']:
+                if attr['type'] == attribute_name:
+                    value = attr['value']
+                for attr2 in o['Attribute']:
```
Comment (cvandeplas) [inline misp_modules/modules/expansion/censys_enrich.py:200]: What is `attr2` ?

**Comment by cvandeplas** [comment]
I've tested the thing, it looks nice ! 
The only question is what `lgtm` raised: the `attr2` variable. 

As discussed we might want to make enhancements to some of the object models, such as the `x509` where many fields are defined as `text`. Please feel free to do another PR for this on the misp-objects repository.
Thanks

**Comment by cvandeplas** [comment]
Thank you very much!

**Comment by adulau** [comment]
Thanks a lot for the contribution. it would be cool to add the JSON with the documentation for the module to have it officially listed on the misp module page.

**Comment by cvandeplas** [comment]
as discussed, this was done and fixed with commit be2786990346f8f5b12812cc0572bf51ba291f7c

**Pull request merged**